db/database.go

```diff
@@ -1826,7 +1826,11 @@ func (db *DatabaseCollectionWithUser) resyncDocument(ctx context.Context, docid,
 		opts := &sgbucket.MutateInOptions{
 			MacroExpansion: macroExpandSpec(base.SyncXattrName),
 		}
-		_, err = db.dataStore.WriteUpdateWithXattrs(ctx, key, []string{base.SyncXattrName, db.userXattrKey()}, 0, nil, opts, writeUpdateFunc)
+		xattrKeys := []string{base.SyncXattrName}
+		if db.userXattrKey() != "" {
+			xattrKeys = append(xattrKeys, db.userXattrKey())
+		}
```

Comment:
A helper function to build the xattr keys wouldn't be a bad idea, since we're going to need to enhance it to add _vv in the near future.